### PR TITLE
Remove generated proto files

### PR DESCRIPTION
**Pull request by dabacon**
Keeps the proto definitions themselves in Cirq.  

Also a variety of cleanup of a pattern where a method both mutated and returned an object.  This was slightly useful for the old protobuf python interface (which was clunky), but is not as useful when just using straight dictionaries.

### CODE REVIEW

File: cirq/google/engine/engine_test.py [hunk 1]
```diff
@@ -13,19 +13,65 @@
...
+            'measurementKeys': [
+                {
+                    'key': 'q',
+                    'qubits': [{'row': 1, 'col': 1}]
+                }
+            ],
+            'parameterizedResults': [
+                {
+                    'params': {'assignments': {'a': 1}},
+                    'measurementResults': base64.b64encode(b'01')
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/google/engine/engine_test.py]: No.  In the proto it gets sents as bytes.  This code converts the binary bit value to a base64 encoded string which is pretty much the optimal encoding modulo some wasted space at end of string and compression.

File: cirq/google/engine/engine.py [hunk 2]
```diff
@@ -348,14 +346,13 @@ def run_sweep(self,
...
         sweeps = _sweepable_to_sweeps(params or ParamResolver({}))
-        proto_program = program_pb2.Program()
-        for sweep in sweeps:
-            sweep_proto = proto_program.parameter_sweeps.add()
-            sweep_to_proto(sweep, sweep_proto)
-            sweep_proto.repetitions = repetitions
-        program_dict = MessageToDict(proto_program)
-        program_dict['operations'] = [MessageToDict(op) for op in
-                                      schedule_to_proto(schedule)]
+        program_dict = {}  # type: Dict
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/google/engine/engine.py]: Done

File: cirq/google/xmon_gates_test.py [hunk 3]
```diff
@@ -14,31 +14,33 @@
...
-from cirq.api.google.v1 import operations_pb2
 from cirq.devices import GridQubit
 from cirq.study import ParamResolver
 from cirq.value import Symbol
 
 
-def proto_matches_text(proto: message, expected_as_text: str):
-    expected = text_format.Merge(expected_as_text, type(proto)())
-    return str(proto) == str(expected)
+Q1 = GridQubit(2, 3)
```
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/google/xmon_gates_test.py]: Done

File: cirq/google/engine/engine_test.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dabacon) [reply cirq/google/engine/engine_test.py]: The proto gets sent as bytes.  You pass a base64 string representation of these bytes to the library.  That's pretty much optimal modulo some terminal wasted bits.

[the remark this answers is not in the document]
Comment (Strilanc) [reply cirq/google/engine/engine_test.py]: 👍

File: cirq/google/xmon_gates_test.py [hunk 4]
```diff
@@ -574,19 +576,21 @@ def test_unsupported_op():
...
+        cg.ExpZGate(half_turns=0.5).to_proto_dict(cirq.GridQubit(2, 3),
+                                                  cirq.GridQubit(3, 4))
     with pytest.raises(ValueError):
-        cg.ExpWGate(half_turns=0.5, axis_half_turns=0).to_proto_dict(Q1, Q2)
+        cg.ExpWGate(half_turns=0.5, axis_half_turns=0).to_proto_dict(
+            cirq.GridQubit(2, 3), cirq.GridQubit(3, 4))
 
 
-def test_symbol_diagrams():
+def test_cirq_sumbol_diagrams():
```
Comment (Strilanc) [inline cirq/google/xmon_gates_test.py]: sumbol -> symbal

...wait

File: docs/development.md [hunk 5]
```diff
@@ -82,13 +82,6 @@ See the previous section for instructions.
     cat apt-dev-requirements.txt apt-runtime-requirements.txt | xargs sudo apt-get install --yes
     ```
 
-    Unfortunately, as of this writing, v3.5 of the [protobuf compiler](https://github.com/google/protobuf) is not installable via `apt-get`.
```
Comment (Strilanc) [inline docs/development.md]: Yesssssss

**Comment by Strilanc** [comment]
Very happy to see this finally going away.

**Pull request merged**